shared/sszutil/deep_equal.go

```diff
@@ -115,6 +117,12 @@ func deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool, depth int) boo
 		}
 		return true
 	case reflect.Uint64:
+		switch v1.Type().Name() {
+		case "Epoch":
+			return v1.Interface().(types.Epoch) == v2.Interface().(types.Epoch)
```

Comment:
For deep value assertions to work shouldn't `uint64` and `types.Epoch` be equal ? This makes it only equal if they are of `types.Epoch`

Reply:
Well, if we are comparing epochs, both should be epochs, imo. While underlying type of Epoch is uint64, and their integral values are the same, I believe if sth is epoch, it should be typed correctly, or not be considered as equal.

Reply:
Fair enough, the deep equal is only used in our pending queue while comparing the same type, so not a big issue. Its fine in that case.